2001-02-14  Jakub Jelinek  <jakub@redhat.com>

	* typeck.c (build_unary_op): Clarify error message.

diff -up gcc/cp/typeck.c.c++-taking-address-error gcc/cp/typeck.c
--- gcc/cp/typeck.c.c++-taking-address-error	2008-07-15 11:09:16.000000000 -0400
+++ gcc/cp/typeck.c	2008-07-15 11:11:15.000000000 -0400
@@ -4662,7 +4662,7 @@ build_unary_op (code, xarg, noconvert)
 	      if (current_class_type
 		  && TREE_OPERAND (arg, 0) == current_class_ref)
 		/* An expression like &memfn.  */
-		cp_pedwarn ("ISO C++ forbids taking the address of a non-static member function to form a pointer to member function.  Say `&%T::%D'", base, name);
+		cp_pedwarn ("ISO C++ forbids taking the address of an unqualified non-static member function to form a pointer to member function.  Say `&%T::%D'", base, name);
 	      else
 		cp_pedwarn ("ISO C++ forbids taking the address of a bound member function to form a pointer to member function", base, name);
 	    }
